From 141f8bc24401db636c4b1a9dd991c68d7b3282f0 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Fri, 31 Jan 2020 19:27:02 +0100
Subject: [PATCH] configure: Use pkg-config to look for keyutils

Support for pkg-config was added to keyutils somewhat recently in
November 2018 [1], whereas the keyutils dependency was originally added
in August 2016 in commit 31ed6f2b3f1ab45ae07aad41c13a51ba91fd159d.

[1] Keyutils commit d4d758a04616b770
    https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git/commit/?id=d4d758a04616b770

https://gitlab.gnome.org/GNOME/gdm/merge_requests/91
---
 configure.ac | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

Index: gdm-3.34.1/configure.ac
===================================================================
--- gdm-3.34.1.orig/configure.ac
+++ gdm-3.34.1/configure.ac
@@ -532,11 +532,10 @@ if test "x$supports_pam_extensions" = "x
 	AC_DEFINE(SUPPORTS_PAM_EXTENSIONS, 1, [Define if PAM supports GDMs custom extensions])
 fi
 
-AC_CHECK_LIB(keyutils, keyctl_read, [
-        AC_DEFINE(HAVE_KEYUTILS, 1, [Define if have keyutils])
-        KEYUTILS_LIBS="-lkeyutils"
-        KEYUTILS_CFLAGS=""
-])
+PKG_CHECK_MODULES([KEYUTILS],
+                  [libkeyutils],
+                  [AC_DEFINE([HAVE_KEYUTILS], [1], [Define if have keyutils])],
+                  [true])
 AC_SUBST(KEYUTILS_LIBS)
 AC_SUBST(KEYUTILS_CFLAGS)
 
